tidy3d/components/lumped_element.py

```diff
@@ -133,7 +168,57 @@ def to_mesh_overrides(self) -> list[MeshOverrideStructure]:
             )
         return mesh_overrides
 
-    @pydantic.validator("voltage_axis", always=True)
+    def to_geometry(self, grid: Grid = None) -> Box:
+        """Converts the :class:`RectangularLumpedElement` object to a :class:`Box`."""
+        box = Box(size=self.size, center=self.center)
+        if grid and self.snap_perimeter_to_grid:
+            return snap_box_to_grid(grid, box, self._snapping_spec)
```

Comment:
Not sure if I missed this earlier, but also, are lumped elements handled differently than 2D materials? Why can’t we directly include them in the structure list and then afterwards construct 2d equivalent and finally volumetric equivalent in simulation postprocessing?

Reply:
Yes, in effect the lumped element will be snapped twice to the grid. I need to perform it earlier, so that I can correctly setup ports in the `TerminalComponentModeler`. There are a lot of other things that depend on the precise location of the lumped element (source/monitors/path integrals).

Also yes, I believe the lumped elements behave exactly like a structure composed of a 2D medium, but just includes some additional snapping behavior right before being converted to a `Structure`. That being said, I think it is possible that the behavior of lumped elements and 2D medium might differ in the future. Also in order to maintain backwards compatibility, we can't really change it at this time.